Dockerfile

```
@@ -15,18 +15,23 @@ COPY . .
 RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o robofuse ./cmd/robofuse
 
 # Final stage - minimal runtime image
-FROM alpine:3.19
+FROM alpine:3.21@sha256:a8560b36e8b8210634f77d9f7f9efd7ffa463e380b75e2e74aff4511df3ef88c
 
-# Install minimal dependencies
-RUN apk --no-cache add ca-certificates tzdata
+# Install minimal dependencies and create non-root user
+RUN apk --no-cache add ca-certificates tzdata && \
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)